File: project/backend-services/smart-irrigation-backend/infrastructure/endpoint/graphql/src/main/java/pt/sensae/services/smart/irrigation/backend/infrastructure/endpoint/graphql/model/device/DeviceLedgerHistoryEntry.java
package pt.sensae.services.smart.irrigation.backend.infrastructure.endpoint.graphql.model.device;

import pt.sensae.services.smart.irrigation.backend.infrastructure.endpoint.graphql.model.data.GPSDataDetails;
import pt.sensae.services.smart.irrigation.backend.infrastructure.endpoint.graphql.model.data.SensorDataHistoryDetails;
import pt.sensae.services.smart.irrigation.backend.infrastructure.endpoint.graphql.model.device.RecordEntry;

import java.util.Set;

public record DeviceLedgerHistoryEntry(String name, GPSDataDetails gps, Set<RecordEntry> records,
                                       Set<? extends SensorDataHistoryDetails> data) {
}

File: project/frontend-services/libs/smart-irrigation/model/src/lib/model/history/DeviceLedgerHistoryEntry.ts
import {GPSDataDetails} from "../details/GPSDataDetails";
import {RecordEntry} from "../device/RecordEntry";
import {SensorDataHistoryDetails} from "./SensorDataHistoryDetails";

export class DeviceLedgerHistoryEntry {
  constructor(public name: string, public gps: GPSDataDetails, public records: Array<RecordEntry>, public data: Array<SensorDataHistoryDetails>) {
  }
}
